sql/lease.go

```diff
@@ -224,14 +227,14 @@ func (s LeaseStore) Publish(tableID ID, update func(*TableDescriptor) error) *ro
 	for r := retry.Start(retryOpts); r.Next(); {
 		// Wait until there are no unexpired leases on the previous version
 		// of the table.
-		expectedVersion, pErr := s.waitForOneVersion(tableID, retryOpts)
-		if pErr != nil {
-			return pErr
+		expectedVersion, err := s.waitForOneVersion(tableID, retryOpts)
+		if err != nil {
+			return roachpb.NewError(err)
 		}
 
 		// There should be only one version of the descriptor, but it's
 		// a race now to update to the next version.
-		pErr = s.db.Txn(func(txn *client.Txn) *roachpb.Error {
+		pErr := s.db.Txn(func(txn *client.Txn) *roachpb.Error {
```

Comment:
This one has the following call flow:

```
   SchemaChangeManager.Start
-> SchemaChanger.exec
-> SchemaChanger.MaybeIncrementVersion
-> LeaseStore.Publish
```

and `pErr.GetDetail()` is called inside `Start`. So, I think we need to pass `roachpb.Error`.